ios/listdevices.go

```diff
@@ -102,9 +103,47 @@ func (muxConn *UsbMuxConnection) ListDevices() (DeviceList, error) {
 	if err != nil {
 		return DeviceList{}, fmt.Errorf("Failed getting devicelist: %v", err)
 	}
+	muxResponse := MuxResponsefromBytes(response.Payload)
+	if muxResponse.Number == 4294967295 {
+		// get list of devices from Listen
+		// this is a workaround for the case where connected devices more than 34
+		return muxConn.ListDevicesFromListen()
+	}
 	return DeviceListfromBytes(response.Payload), nil
 }
 
+func (muxConn *UsbMuxConnection) ListDevicesFromListen() (DeviceList, error) {
+	attachedReceiver, err := muxConn.Listen()
+	if err != nil {
+		return DeviceList{}, fmt.Errorf("Failed to start listening: %v", err)
+	}
+	defer muxConn.Close()
+
+	deviceEntryChan := make(chan DeviceEntry)
+	go func() {
+		for {
+			msg, err := attachedReceiver()
+			if err != nil {
+				break
+			}
+			if msg.DeviceAttached() {
+				deviceEntryChan <- msg.DeviceEntry()
+			}
+		}
+	}()
+	var deviceList = DeviceList{
+		DeviceList: make([]DeviceEntry, 0, 40),
+	}
+	for {
+		select {
+		case entry := <-deviceEntryChan:
+			deviceList.DeviceList = append(deviceList.DeviceList, entry)
+		case <-time.After(100 * time.Millisecond):
+			return deviceList, nil
```

Comment:
is 100ms the only way to know we have reached the last device? such absolute timing can be random when dealing with services. what do you think?

Reply:
em.. do you have some better way?